src/accelerate/utils/dataclasses.py

```diff
@@ -1197,147 +1196,230 @@ class FullyShardedDataParallelPlugin:
+            ShardingStrategy,
+            StateDictType,
+        )
 
-        prefix = "FSDP_"
+        env_prefix = "FSDP_"
+        # Strategy: By default we should always assume that values are passed in, else we check the environment variables
         if self.sharding_strategy is None:
-            sharding_strategy = os.environ.get(prefix + "SHARDING_STRATEGY", "FULL_SHARD")
-            sharding_strategy = (
-                FSDP_SHARDING_STRATEGY.index(sharding_strategy) + 1
-                if not sharding_strategy.isdigit()
-                else int(sharding_strategy)
-            )
-            self.sharding_strategy = ShardingStrategy(sharding_strategy)
+            self.sharding_strategy = os.environ.get(env_prefix + "SHARDING_STRATEGY", "FULL_SHARD")
+        if isinstance(self.sharding_strategy, str):
+            # We need to remap based on custom enum values for user readability
+            if self.sharding_strategy.upper() in FSDP_SHARDING_STRATEGY:
+                self.sharding_strategy = FSDP_SHARDING_STRATEGY.index(self.sharding_strategy.upper()) + 1
+            if isinstance(self.sharding_strategy, int) or self.sharding_strategy.isdigit():
+                self.sharding_strategy = ShardingStrategy(int(self.sharding_strategy))
+            else:
+                self.sharding_strategy = ShardingStrategy[self.sharding_strategy.upper()]
 
         if self.cpu_offload is None:
-            if str_to_bool(os.environ.get(prefix + "OFFLOAD_PARAMS", "False")) == 1:
-                self.cpu_offload = CPUOffload(offload_params=True)
-            else:
-                self.cpu_offload = CPUOffload(offload_params=False)
+            self.cpu_offload = str_to_bool(os.environ.get(env_prefix + "OFFLOAD_PARAMS", "False")) == 1
+        if isinstance(self.cpu_offload, bool):
+            self.cpu_offload = CPUOffload(offload_params=self.cpu_offload)
 
         if self.backward_prefetch is None:
-            prefetch_policy = os.environ.get(prefix + "BACKWARD_PREFETCH", "NO_PREFETCH")
-            if prefetch_policy != FSDP_BACKWARD_PREFETCH[-1]:
-                self.backward_prefetch = BackwardPrefetch(FSDP_BACKWARD_PREFETCH.index(prefetch_policy) + 1)
+            self.backward_prefetch = os.environ.get(env_prefix + "BACKWARD_PREFETCH", "NO_PREFETCH")
+        if isinstance(self.backward_prefetch, str):
+            if self.backward_prefetch.upper() == "NO_PREFETCH":
+                self.backward_prefetch = None
+            else:
+                if self.backward_prefetch in FSDP_BACKWARD_PREFETCH:
+                    self.backward_prefetch = FSDP_BACKWARD_PREFETCH.index(self.backward_prefetch.upper()) + 1
+                if isinstance(self.backward_prefetch, int) or self.backward_prefetch.isdigit():
+                    self.backward_prefetch = BackwardPrefetch(int(self.backward_prefetch))
+                else:
+                    self.backward_prefetch = BackwardPrefetch[self.backward_prefetch.upper()]
```

Comment:
Could this be simplified or am I missing something?

```suggestion
            self.backward_prefetch = os.environ.get(env_prefix + "BACKWARD_PREFETCH", None)
        if isinstance(self.backward_prefetch, str):
            if self.backward_prefetch in FSDP_BACKWARD_PREFETCH:
                self.backward_prefetch = FSDP_BACKWARD_PREFETCH.index(self.backward_prefetch.upper()) + 1
            if isinstance(self.backward_prefetch, int) or self.backward_prefetch.isdigit():
                self.backward_prefetch = BackwardPrefetch(int(self.backward_prefetch))
            else:
                self.backward_prefetch = BackwardPrefetch[self.backward_prefetch.upper()]
```

Reply:
You're missing the fact that `NO_PREFETCH` is in `FSDP_BACKWARD_PREFETCH`

Reply:
So this should work?

```python
            self.backward_prefetch = os.environ.get(env_prefix + "BACKWARD_PREFETCH", None)
                if isinstance(self.backward_prefetch, str) and (self.backward_prefetch != "NO_PREFETCH"):
	            if self.backward_prefetch in FSDP_BACKWARD_PREFETCH:
	                self.backward_prefetch = FSDP_BACKWARD_PREFETCH.index(self.backward_prefetch.upper()) + 1
	            if isinstance(self.backward_prefetch, int) or self.backward_prefetch.isdigit():
	                self.backward_prefetch = BackwardPrefetch(int(self.backward_prefetch))
	            else:
	                self.backward_prefetch = BackwardPrefetch[self.backward_prefetch.upper()]
```

My main concern is to avoid setting `self.backward_prefetch` from `None` to `"NO_PREFETCH"` back to `None`, which is confusing.